[23.05] kernel: bump 5.15 to 5.15.198

sch-m opened the pull request:
@hauke: I've updated linux patchlevel to v5.15.196 und refreshed all patches for all targets.

PolynomialDivision committed:
tools/zlib: update to 1.3

Changes in 1.3 (18 Aug 2023)
- Remove K&R function definitions and zlib2ansi
- Fix bug in deflateBound() for level 0 and memLevel 9
- Fix bug when gzungetc() is used immediately after gzopen()
- Fix bug when using gzflush() with a very small buffer
- Fix crash when gzsetparams() attempted for transparent write
- Fix test/example.c to work with FORCE_STORED
- Rewrite of zran in examples (see zran.c version history)
- Fix minizip to allow it to open an empty zip file
- Fix reading disk number start on zip64 files in minizip
- Fix logic error in minizip argument processing
- Add minizip testing to Makefile
- Read multiple bytes instead of byte-by-byte in minizip unzip.c
- Add memory sanitizer to configure (--memory)
- Various portability improvements
- Various documentation improvements
- Various spelling and typo corrections

Signed-off-by: Nick Hainke <vincent@systemli.org>
(cherry picked from commit 63f5b5ed3f4a23e78d72b38c18c76409190a5c14)

PolynomialDivision committed:
tools/zlib: update to 1.3.1

Release Notes:
https://github.com/madler/zlib/releases/tag/v1.3.1

Signed-off-by: Nick Hainke <vincent@systemli.org>
(cherry picked from commit 25080a95dccff758643bca5cc60fec304f872141)

sch-m committed:
kernel: bump 5.15 to 5.15.196

Removed because they are upstream:
   bcm27xx/patches-5.15/950-0119-media-tc358743-Increase-FIFO-level-to-374.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=aef1b717d4a27e72f9c1d1ebe32c6d841a7e69fc

   bcm27xx/patches-5.15/950-0122-media-tc358743-Check-I2C-succeeded-during-probe.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=a6ccbe037734ef20016bc57869a90abc4406ea7e

   bcm27xx/patches-5.15/950-0128-media-tc358743-Return-an-appropriate-colorspace-from.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=21ba26a8e3477998df9f52a519169b4c6764c0b2

   bcm27xx/patches-5.15/950-0278-media-i2c-tc358743-Only-allow-supported-pixel-fmts-i.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=21ba26a8e3477998df9f52a519169b4c6764c0b2

   generic/backport-5.15/895-v6.15-0001-sch_htb-make-htb_qlen_notify-idempotent.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=967955c9e57f8eebfccc298037d4aaf3d42bc1c9

   generic/backport-5.15/895-v6.15-0002-sch_drr-make-drr_qlen_notify-idempotent.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=db568d2151cd20a1e516b0454eda8a11a948e72a

   generic/backport-5.15/895-v6.15-0003-sch_hfsc-make-hfsc_qlen_notify-idempotent.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=a5efc95a33bd4fcb879250852828cc58c7862970

   generic/backport-5.15/895-v6.15-0004-sch_qfq-make-qfq_qlen_notify-idempotent.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=44e2f93f9820a54929ab8cfc6d322a18279bb79b

   generic/backport-5.15/895-v6.15-0005-sch_htb-make-htb_deactivate-idempotent.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=c2d25fddd867ce20a266806634eeeb5c30cb520c

   mediatek/patches-5.15/821-v5.16-pinctrl-mediatek-moore-check-if-pin_desc-is-valid-be.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=4b5022b649abbceb564253fafa9d2c7914d29ee4

Manually adapted:
   bcm53xx/patches-5.15/180-usb-xhci-add-support-for-performing-fake-doorbell.patch
   upstream changes to the xhci_free_virt_device():
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=c78c8e5048b7069b6655613ab4e9795a901dbbee
   Fix taken from commit efd444537dec. Thanks to Goetz Goerisch.

   generic/backport-5.15/882-v5.19-rndis_host-limit-scope-of-bogus-MAC-address-detectio.patch
   upstream changes to generic_rndis_bind():
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=a6b33d9edf2600bc9647e4f0420714beae6eb329

   lantiq/patches-5.15/0152-lantiq-VPE.patch
   upstream changes to arch/mips/include/asm/vpe.h:
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=90c1295da0a78f3ed00388bb5f569d843b18c03d

Adapt kernel configuration for newly added option
   generic: add CONFIG_RESET_BRCMSTB is not set
   upstream changes to drivers/reset/Kconfig:
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=345df19a971a506813d7a33106e35aac31489e78

   x86: add CONFIG_MITIGATION_VMSCAPE=y
   upstream changes to arch/x86/Kconfig:
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=2f4f2f8f860cb4c3336a7435ebe8dcfded0c9c6e
   (cherry picked from commit 6af8e20b45d7)

Signed-off-by: Martin Schiller <ms@dev.tdt.de>

GoetzGoerisch commented:
You need to fix the bcm53xx patch similar to https://github.com/openwrt/openwrt/commit/efd444537dec1bb3f222fc4db859edaab5a102f0

sch-m committed:
kernel: bump 5.15 to 5.15.197

Removed because they are upstream:
   bcm27xx/patches-5.15/950-0030-Revert-Bluetooth-Always-request-for-user-confirmatio.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=4475bac8224ce1e43603a94411ab3d781f006326

   bcm27xx/patches-5.15/950-0031-Revert-Bluetooth-Always-request-for-user-confirmatio.patch
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=4475bac8224ce1e43603a94411ab3d781f006326

Signed-off-by: Martin Schiller <ms@dev.tdt.de>

sch-m committed:
tools/gnulib: add patch to fix build on macos

This backports a fix for lib/posixtm.c.

Signed-off-by: Martin Schiller <ms@dev.tdt.de>

sch-m commented:
@hauke All CI checks are green. Is there any chance to get this kernel version bump for 23.05 merged?

sch-m committed:
kernel: bump 5.15 to 5.15.198

Manually adapted:
   generic/backport-5.15/601-v5.18-page_pool-Add-recycle-stats.patch
   upstream changes to page_pool_recycle_in_ring():
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=d69f28ef7cdafdcf37ee310f38b1399e7d05f9a8

   ipq40xx/patches-5.15/401-mmc-sdhci-msm-comment-unused-sdhci_msm_set_clock.patch
   upstream changes to sdhci_msm_set_clock():
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=d685237855bf0f27fe2ac9c321374ead30e744c8

   ipq807x/patches-5.15/0113-remoteproc-qcom-Add-secure-PIL-support.patch
   upstream changes to MAX_HALT_REG
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=140e6cd7b092525c151f15952cca5b8df9dcc74a

   ipq807x/patches-5.15/0116-remoteproc-qcom-Update-regmap-offsets-for-halt-regis.patch
   upstream changes to MAX_HALT_REG
   https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.15.y&id=140e6cd7b092525c151f15952cca5b8df9dcc74a

Signed-off-by: Martin Schiller <ms@dev.tdt.de>
